fetch2/src/main/java/com/tonyodev/fetch2/DownloadManager.java

```diff
@@ -54,6 +54,30 @@ void pause(long id) {
         actionProcessor.processNext();
     }
 
+    void pause(final String groupId) {
```

Comment:
What happens when you pause a download that is queued? I would expect to be able to pause a queued download. If a queued download is paused, it should not start when it is at the front of the queue. This case is especially applicable to batches. It doesn't look like the pause method actually updates the database directly, it only updates it when an interrupt is detected in the `run()` method